src/normalizer.ts

```diff
@@ -151,17 +151,57 @@ export const normalizeHexOrComplexColor = (
 
 export const normalizeText = (t: TextChild): InternalTextPart[] => {
   if (isReactElementOrElementArray(t)) {
-    return flattenChildren(t).map(
+    return flattenChildren(t).reduce<InternalTextPart[]>(
       (
+        textParts,
         el:
           | string
           | number
           | ReactElement<TextLinkProps>
           | ReactElement<TextBulletProps>
       ) => {
         if (React.isValidElement(el)) {
-          let link;
           let bullet;
+          if (isTextBullet(el)) {
+            // We know the intention is for a bullet, so pass on true if no customisation required
+            const { children, style, rtlMode, lang, ...bulletProps } = el.props;
+            bullet = Object.keys(bulletProps).length ? bulletProps : true;
+
+            // If the bullet has multiple TextChild components, then we make
+            // the first one take on `bullet` prop and collect the rest up.
+            let childParts: InternalTextPart[] = [];
+            if (Array.isArray(children)) {
+              childParts = children
+                .map((child) => normalizeText(child))
+                .reduce((prev, current) => prev.concat(current), []);
+            } else {
+              childParts = normalizeText(children);
+            }
+
+            const color = childParts[0].style?.color ?? style?.color;
+            // Make `breakLine = false` for all child components except the last one
+            // (so every child will sit within the same bullet point)
+            const firstChild = {
+              rtlMode,
+              lang,
+              ...childParts[0],
+              bullet,
+              breakLine: childParts.length <= 1,
+              style: {
+                ...(style || {}),
+                ...(childParts[0].style || {}),
+                color: color ? normalizeHexColor(color) : undefined,
```

Comment:
The other overrides/underrides (😁) here seem OK, but I'm wondering about the color one. If the `child.style.color` was overridden already before normalization up there on line 175, the subtle but still existing repeating code here could be omitted.

This change doesn't make much sense to do though if the `style` only applying to the first node behavior is retained

Reply:
I think its a good idea, and did initially try it that way, but then realised that normalising them all first helps to be able to treat strings/numbers/react elements in the same way (objects with styles). Does the latest commit meet what you were thinking? https://github.com/wyozi/react-pptx/pull/414/commits/540d2a813c49b681c95156465cbc71505bdfd461

Reply:
Yeah, looks good!